lib/routes/compact.js

```diff
@@ -7,11 +7,17 @@ module.exports = function (app) {
 
   // DB Compaction
   app.post('/:db/_compact', utils.jsonParser, function (req, res, next) {
-    req.db.compact(utils.makeOpts(req), function (err, response) {
-      if (err) {
-        return utils.sendError(res, err);
-      }
-      utils.sendJSON(res, 200, {ok: true});
+    // don't wait for compaction to finish, but do give errors a chance
+    // to pop up by waiting for a little bit. Yes, this is a hack.
+    var encounteredError = false;
+    req.db.compact(utils.makeOpts(req)).catch(function (err) {
+      encounteredError = true;
+      return utils.sendError(res, err);
     });
+    setTimeout(function () {
+      if (!encounteredError) {
+        utils.sendJSON(res, 202, {ok: true});
+      }
+    }, 0);
```

Comment:
CouchDB doesn't always return ``ok: true``: db ``/_security`` can make it throw errors. (In fact, couchdb-harness tests that IIRC).

As for why I don't wait for compaction completion as it did: on massive databases it might take a while. That's no problem for express, but couchdb-harness (and other couchdb tools probably) expects a fast return and 202. So it's for couchdb compatibility.